pkg/http/api.go

```diff
@@ -40,31 +40,34 @@ type api struct {
 	pubSub                pubsub.PubSub
 	sendToOutputBindingFn func(name string, req *bindings.WriteRequest) error
 	id                    string
+	metadata              map[string]string
 }
 
 type metadata struct {
 	ID                string                     `json:"id"`
 	ActiveActorsCount []actors.ActiveActorsCount `json:"actors"`
+	AppCommand        string                     `json:"appCommand"`
 }
 
 const (
-	apiVersionV1         = "v1.0"
-	idParam              = "id"
-	methodParam          = "method"
-	actorTypeParam       = "actorType"
-	actorIDParam         = "actorId"
-	storeNameParam       = "storeName"
-	stateKeyParam        = "key"
-	secretStoreNameParam = "secretStoreName"
-	secretNameParam      = "key"
-	topicParam           = "topic"
-	nameParam            = "name"
-	consistencyParam     = "consistency"
-	retryIntervalParam   = "retryInterval"
-	retryPatternParam    = "retryPattern"
-	retryThresholdParam  = "retryThreshold"
-	concurrencyParam     = "concurrency"
-	daprSeparator        = "||"
+	apiVersionV1          = "v1.0"
+	idParam               = "id"
+	methodParam           = "method"
+	actorTypeParam        = "actorType"
+	actorIDParam          = "actorId"
+	storeNameParam        = "storeName"
+	stateKeyParam         = "key"
+	secretStoreNameParam  = "secretStoreName"
+	secretNameParam       = "key"
+	topicParam            = "topic"
+	nameParam             = "name"
+	consistencyParam      = "consistency"
+	retryIntervalParam    = "retryInterval"
+	retryPatternParam     = "retryPattern"
+	retryThresholdParam   = "retryThreshold"
+	concurrencyParam      = "concurrency"
+	daprSeparator         = "||"
+	metadataKeyAppCommand = "appCommand"
```

Comment:
If we're introducing a change that makes metadata mutable, it would be great to allows for arbitrary key/values to be places in there and not have the runtime know specifically about each and every command.

It's preferable in terms of maintenance, and allows the runtime to only know about metadata that it actually cares about, meaning related to the runtime and not to externals.

So I would prefer to see something like a `Custom` map or something along those lines to account for it..

Reply:
I can change the name and remove the "whitelist" of acceptable keys. Would it work?

Reply:
Made the change. It is a map called "extended" in the metadata type.